vmas/scenarios/transport.py

```diff
@@ -22,27 +22,43 @@
 class Scenario(BaseScenario):
     def make_world(self, batch_dim: int, device: torch.device, **kwargs):
         n_agents = kwargs.get("n_agents", 4)
+
+        # packages
         self.n_packages = kwargs.get("n_packages", 1)
         self.package_width = kwargs.get("package_width", 0.15)
         self.package_length = kwargs.get("package_length", 0.15)
-        self.package_observation_radius = kwargs.get("package_observation_radius", 0.35)
+
+        # partial obs
         self.partial_observations = kwargs.get("partial_observations", True)
+        self.package_observation_radius = kwargs.get("package_observation_radius", 0.35)
 
+        # realism
+        self.linear_friction = kwargs.get("linear_friction", 0.1)
+        self.angular_friction = kwargs.get("angular_friction", 0.1)
+        self.drag = kwargs.get("drag", 0.0)
         self.package_mass = kwargs.get("package_mass", 50)
-        
-        self.add_dense_reward = kwargs.get("add_dense_reward", True)
+        # TODO: implement automated domain randomization here?
+
+        # rewards
         self.agent_package_dist_reward_factor = kwargs.get("agent_package_dist_reward_factor", 0.1)
         self.package_goal_dist_reward_factor = kwargs.get("package_goal_dist_reward_factor", 100)
+        self.interagent_collision_penalty = kwargs.get("interagent_collision_penalty", -1)
+        assert self.interagent_collision_penalty <= 0, f"self.interagent_collision_penalty must be <= 0, current value is {self.interagent_collision_penalty}!"
+        self.add_dense_reward = kwargs.get("add_dense_reward", True)
         self.package_on_goal_reward_factor = kwargs.get("package_on_goal_reward_factor", 1.0)
+
+        # capabilities
         self.capability_mult_range = kwargs.get("capability_mult_range", [0.5, 2])
         self.capability_mult_min = self.capability_mult_range[0]
         self.capability_mult_max = self.capability_mult_range[1]
         self.capability_representation = kwargs.get("capability_representation", "raw")
 
+        # general world settings
         self.world_semidim = 0.75 
         self.default_agent_radius = 0.03
         self.default_agent_u = 0.6
         self.default_agent_mass = 1.0
+        self.min_collision_distance = 0.05 * self.default_agent_radius # default navigation collision dist is 5% of the agent radius
```

Comment:
Are agent collisions determined based on distance from center to center of agents, or perimeter to perimeter. If it is the former, then agents would actually visually show collision before a collision is actually registered. This should be checked. If it is the case that it is determined from the center of robots, maybe make it 1.05 agents default radius.

Reply:
Perimeter to perimeter, see the collision checking code here: https://github.com/GT-STAR-Lab/VectorizedMultiAgentSimulator/blob/main/vmas/simulator/core.py#L1419-L1421

Additionally, I largely ripped the collision distance code from VMAS' navigation environment, where it seems to work for penalizing collisions there.